lib/std/math/test/exp.zig

```diff
@@ -0,0 +1,131 @@
+    const cases = test_utils.genTests(Tc32, special_tests) ++ test_utils.nanTests(Tc32);
+    try runTests(cases);
+}
+
+test "math.exp32() boundary" {
+    const cases = [_]Tc32{
+        // zig fmt: off
+        tc32( 0x1.62e42ep+6,   0x1.ffff08p+127), // The last value before the result gets infinite
+        tc32( 0x1.62e430p+6,   inf32          ), // The first value that gives inf
+        tc32( 0x1.fffffep+127, inf32          ), // Max input value
+        tc32( 0x1p-149,        1              ), // Min positive input value
+        tc32(-0x1p-149,        1              ), // Min negative input value
+        tc32( 0x1p-126,        1              ), // First positive subnormal input
+        tc32(-0x1p-126,        1              ), // First negative subnormal input
+        tc32(-0x1.9fe368p+6,   0x1p-149       ), // The last value before the result flushes to zero
+        tc32(-0x1.9fe36ap+6,   0              ), // The first value at which the result flushes to zero
+        tc32(-0x1.5d589ep+6,   0x1.00004cp-126), // The last value before the result flushes to subnormal
+        tc32(-0x1.5d58a0p+6,   0x1.ffff98p-127), // The first value for which the result flushes to subnormal
+        // zig fmt: on
+    };
+    try runTests(cases);
+}
+
+test "math.exp64() sanity" {
+    const cases = [_]Tc64{
+        // zig fmt: off
+        tc64(-0x1.02239f3c6a8f1p+3, 0x1.490327ea61235p-12),
+        tc64( 0x1.161868e18bc67p+2, 0x1.34712ed238c04p+6 ),
+        tc64(-0x1.0c34b3e01e6e7p+3, 0x1.e06b1b6c18e64p-13),
+        tc64(-0x1.a206f0a19dcc4p+2, 0x1.7dd47f810e68cp-10),
+        tc64( 0x1.288bbb0d6a1e6p+3, 0x1.4abc77496e07ep+13),
+        tc64( 0x1.52efd0cd80497p-1, 0x1.f04a9c1080500p+0 ),
+        tc64(-0x1.a05cc754481d1p-2, 0x1.54f1e0fd3ea0dp-1 ),
+        tc64( 0x1.1f9ef934745cbp-1, 0x1.c0f6266a6a547p+0 ),
+        tc64( 0x1.8c5db097f7442p-1, 0x1.1599b1d4a25fbp+1 ),
+        tc64(-0x1.5b86ea8118a0ep-1, 0x1.03b5728a00229p-1 ),
+        tc64(-0x1.57f25b2b5006dp+2, 0x1.2fbea6a01cab9p-8 ),
+        tc64( 0x1.c7d30fb825911p+3, 0x1.76eeed45a0634p+20),
+        tc64( 0x1.19be709de7505p+4, 0x1.52d3eb7be6844p+25),
+        tc64(-0x1.ab6d6fba96889p+3, 0x1.a88ae12f985d6p-20),
+        tc64(-0x1.5ac18e27084ddp+2, 0x1.22b327da9cca6p-8 ),
+        tc64(-0x1.925981b093c41p-1, 0x1.d2acc046b55f7p-2 ),
+        tc64( 0x1.7221cd18455f5p+3, 0x1.9c2cde8699cfbp+16),
+        tc64( 0x1.11a0d4a51b239p+4, 0x1.980ef612ff182p+24),
+        tc64(-0x1.ae41a1079de4dp+1, 0x1.1c28d16bb3222p-5 ),
+        tc64(-0x1.329153103b871p+4, 0x1.47efa6ddd0d22p-28),
+        // zig fmt: on
+    };
+    try runTests(cases);
+}
+
+test "math.exp64() special" {
+    const cases = test_utils.genTests(Tc64, special_tests) ++ test_utils.nanTests(Tc64);
+    try runTests(cases);
+}
+
+test "math.exp64() boundary" {
+    const cases = [_]Tc64{
+        // zig fmt: off
+        tc64( 0x1.62e42fefa39efp+9,    0x1.fffffffffff2ap+1023), // The last value before the result gets infinite
```

Comment:
when I read stuff like this I am wondering, why no `-inf64`?
Why not sort from min to max input?

Reply:
My intention was the following:
- 'special' tests covering special inputs, e.g. `0`, `1`, `+/-inf`, various NaNs, special numbers such as `ln2` or `e`
- 'boundary' tests covering special outputs (first/last inputs giving results in the different parts of the output range), e.g. last input before result is inf, first input where result is inf, ...
- 'sanity' tests for some other random/regular inputs just to give a sanity check over whether the implementation is working for mainline cases (I don't consider the actual values that important, since the special/boundary tests should be covering the full domain/range)

I ended up also including boundary *input* values (not covered in the 'special' tests) in the 'boundary' tests too though, such as float max/min/true min, which are definitely worth covering (there are code branches for handling subnormal inputs for example). There didn't seem to be much value in including all boundary inputs if they result in the same output (e.g. `exp(x) -> 0`, where `x < -1p+7`, so why bother with `-floatMax()` when checking the first value that hits zero). However, I'm fine with including all those cases more systematically if you think that's worth doing.